internal/services/topic.go

```diff
@@ -80,24 +80,41 @@ func (ts *TopicService) Reconcile() (TopicReconcileResult, error) {
 	topicMetadata := metadata[0]
 
 	if topicMetadata.Err == sarama.ErrUnknownTopicOrPartition {
+
 		// canary topic doesn't exist, going to create it
 		log.Printf("The canary topic %s doesn't exist\n", topicMetadata.Name)
-		if result.Assignments, err = ts.createTopic(len(brokers)); err != nil {
-			log.Printf("Error creating topic %s: %v", topicMetadata.Name, err)
-			return result, err
+		// topic is created if "dynamic" reassignment is enabled or the expected brokers are provided by the describe cluster
+		if ts.canaryConfig.ExpectedClusterSize == config.ExpectedClusterSizeDefault ||
+			ts.canaryConfig.ExpectedClusterSize == len(brokers) {
+
+			if result.Assignments, err = ts.createTopic(len(brokers)); err != nil {
+				log.Printf("Error creating topic %s: %v", topicMetadata.Name, err)
+				return result, err
+			}
+			log.Printf("The canary topic %s was created\n", topicMetadata.Name)
+		} else {
+			// not creating the topic and returning error to avoid starting producer/consumer
+			return result, topicMetadata.Err
```

Comment:
So does this mean we crash loop until the cluster has the expected size? Should we log the reason we're returning here, since the caller won't be able to distinguish this from other errors, will it?

Reply:
Sad to say but yes, right now this is the way it works right now. There are two issues #4 and #5 to address better logging and error handling. Anyway, it makes sense to log a specific message in this case.